ui/raidboss/data/07-dt/ultimate/futures_rewritten.ts

```diff
@@ -481,10 +623,668 @@ const triggerSet: TriggerSet<Data> = {
         unknown: Outputs.unknown,
       },
     },
-    // Crystals
+    {
+      id: 'FRU P2 Shiva Cleave Add Collect',
+      type: 'StartsUsing',
+      // The Shiva that will cast Twin Silence/Twin Stillness is the same actor
+      // that casts Sinbound Holy (9D10).
+      netRegex: { id: '9D10' },
+      run: (data, matches) => {
+        const x = parseFloat(matches.x);
+        const y = parseFloat(matches.y);
+        data.p2KBShivaDir = Directions.xyTo8DirOutput(x, y, centerX, centerY);
+      },
+    },
+    {
+      // TODO: Add 'Always Away, Cursed Clockwise' strat
+      id: 'FRU P2 Sinbound Holy Rotation',
+      type: 'Ability',
+      // We can determine the player's knockbac dir from Heavenly Strike (9D0F).  It occurs after
+      // Shiva is already casting Sinbound Holy.
+      netRegex: { id: '9D0F' },
+      condition: Conditions.targetIsYou(),
+      durationSeconds: 5,
+      infoText: (data, matches, output) => {
+        const x = parseFloat(matches.targetX);
+        const y = parseFloat(matches.targetY);
+        const playerDir = Directions.xyTo8DirOutput(x, y, centerX, centerY);
+        const addDir = data.p2KBShivaDir ?? 'unknown';
+        if (playerDir === 'unknown' || addDir === 'unknown')
+          return;
+
+        const relPos = getRelativeClockPos(playerDir, addDir);
+        if (data.triggerSetConfig.sinboundRotate === 'aacc')
+          switch (relPos) {
+            case 'same':
+            case 'opposite':
+              return output.aaccCursed!();
+            case 'clockwise':
+              return output.aaccClockwise!();
+            case 'counterclockwise':
+              return output.aaccCounterclockwise!();
+            default:
+              break;
+          }
+        return output[relPos]!();
+      },
+      outputStrings: {
+        aaccCursed: {
+          en: 'Cursed Add - Fast Clockwise',
+        },
+        aaccClockwise: {
+          en: 'Rotate Counterclockwise (away from add)',
+        },
+        aaccCounterclockwise: {
+          en: 'Rotate Clockwise (away from add)',
+        },
```

Comment:
Just a drive-by on this one since I don't have time tonight for a full review. Is it maybe worth naming these as `aaccRotate[direction]`, since otherwise it could be slightly confusing to have opposing directions in the config UI?

Reply:
Yeah, that seems much better.